### ISSUE TITLE
PDF files do not occupy 100% of height of the page

### ISSUE THREAD

**Issue by mathwww** [RainLoop/rainloop-webmail#1919]
**RainLoop version**: 1.13.0; **Browser**: Google Chrome 77.0.3865.75 and above; **OS**: Ubuntu 19.04 and Windows 10

**Expected behavior and actual behavior:** Clicking on eye icon by hover a PDF file attached to the email should open the file  and the height page should be regular (100%) the file gets restricted to an amount of percentage of the screen size causing overflow despite of the available size though.

**Steps to reproduce the problem:** Open an attached PDF file by clicking in the eye icon that is shown by hovering mouse through the attachment .

**Logs or screenshots:** The logs show below are given in another browsers either, but only in Chrome they are causing the issue: 

![Captura de tela de 2019-09-20 14-04-52](https://user-images.githubusercontent.com/18701219/65345459-b80b8300-dbb0-11e9-90c8-1cb3d8ba14a0.png)

**Problem:**
![Captura de tela de 2019-09-18 12-18-05](https://user-images.githubusercontent.com/18701219/65345541-e9844e80-dbb0-11e9-99d4-48058a403a7a.png)
![image](https://user-images.githubusercontent.com/18701219/65345624-133d7580-dbb1-11e9-95c9-308e9d34d3e8.png)

**Comment by newfield86** [comment]
This is indeed a annoying issue and its amazing that this ticket is open for more than a month without a fix/response! 😮 Imagine if you paid for a premium version..

**Comment by janvs0** [comment]
It happens when RainLoop sends a CSP header in combination with Chrome inline styling its PDF viewer.

You can work around it by letting RainLoop send a slightly modified header in ServiceActions.php, line 501, the public function ServiceRaw():
```php
@\header('Content-Security-Policy: default-src \'self\'; script-src \'none\'; style-src \'none\'; frame-src \'none\'; child-src \'none\'', true);
```
Change this to:
```php
@\header('Content-Security-Policy: default-src \'self\'; script-src \'none\'; style-src \'unsafe-inline\'; frame-src \'none\'; child-src \'none\'', true);
```
Beware this has security implications.

**Comment by mathwww** [comment]
I guess it would be easier to remove this icon even from chrome since It does not work in other browsers, and It would not take any security issue.

**Comment by HugoCalado** [comment]
I`m facing the same issue.This is indeed annoying.

**Comment by kurim** [comment]
Instead of using unsafe-inline you also can use:

```php
@\header('Content-Security-Policy: default-src \'self\'; script-src \'none\'; style-src \'unsafe-hashes\' \'sha256-C7vpsE1KLI7RuUgCprJTQZin6dWK+ccynbOx+OqjVow=\' \'sha256-tbWZ4NP1341cpcrZVDn7B3o9bt/muXgduILAnC0Zbaw=\'; frame-src \'none\'; child-src \'none\'', true)
```

Based on the bug post for chromium engine:
https://bugs.chromium.org/p/chromium/issues/detail?id=271452#c22

This way is a bit saver than unsave-inline:
https://www.chromestatus.com/feature/5867082285580288

**Comment by janvs0** [comment]
Correct, although you'd need to rehash when chrome changes their pdf viewer.On 4 Nov 2019 20:33, Kurim <notifications@github.com> wrote:Instead of using unsafe-inline you also can use:
@\header('Content-Security-Policy: default-src \'self\'; script-src \'none\'; style-src \'unsafe-hashes\' \'sha256-C7vpsE1KLI7RuUgCprJTQZin6dWK+ccynbOx+OqjVow=\' \'sha256-tbWZ4NP1341cpcrZVDn7B3o9bt/muXgduILAnC0Zbaw=\'; frame-src \'none\'; child-src \'none\'', true)
Based on the bug post for chromium engine:
https://bugs.chromium.org/p/chromium/issues/detail?id=271452#c22
This way is a bit saver than unsave-inline:
https://www.chromestatus.com/feature/5867082285580288

**Comment by mattv8** [comment]
All,

Perhaps a more robust fix is to correct the global Content-Security-Policy in your webserver config. 

First, comment out line 501 in /rainloop/v/1.13.XX/app/libraries/RainLoop/ServiceActions.php (this step may not be necessary if Apache config overrides?):
`//@\header('Content-Security-Policy: default-src \'self\'; script-src \'none\'; style-src \'none\'; frame-src \'none\'; child-src \'none\'', true);`

Then in Apache config, add the section:
`<IfModule mod_headers.c>
    Header set Content-Security-Policy "default-src 'self'; img-src https; script-src 'self' https://apis.google.com"
    Header set X-Content-Type-Options nosniff
    Header set X-Frame-Options DENY
</IfModule>`

Restart Apache. Clear Rainloop cache. Refresh page and test.

This can also be accomplished in IIS and NGINX. Read more about it here: https://ole.michelsen.dk/blog/secure-your-website-with-content-security-policy.html

**Comment by mattv8** [comment]
I am wrong. Somehow I was able to have it working with a global config, then it stopped working after 20 minutes or so. I may have been running off a cached file with disabled CSP's.
